packages/server/src/trpc-routers/__tests_e2e__/swap-router.test.ts

```diff
@@ -0,0 +1,1160 @@
+  expect(isNumeric(amount)).toBeTruthy();
+  // Make sure amount is not negative
+  expect(parseFloat(amount)).toBeGreaterThan(0);
+
+  // Swap fee
+  // expect(reply.swapFee).toBeInstanceOf(RatePretty);
+  // // Should match with the format of "0.1%"
+  // expect(reply.swapFee?.toString()).toMatch(percentageRegex);
+
+  // const swapFee = reply.swapFee!.toDec().toString();
+  // expect(isNumeric(swapFee)).toBeTruthy();
+  // expect(parseFloat(swapFee)).toBeGreaterThan(0);
+
+  // Price impact token out
+  expect(reply.priceImpactTokenOut).toBeInstanceOf(RatePretty);
+  // Should match with the format of "0.1%"
+  expect(reply.priceImpactTokenOut?.toString()).toMatch(percentageRegex);
+
+  const priceImpactTokenOut = reply.priceImpactTokenOut!.toDec().toString();
+  expect(isNumeric(priceImpactTokenOut)).toBeTruthy();
+  expect(parseFloat(priceImpactTokenOut)).toBeGreaterThan(0);
+
+  // Token in fee amount
+  // expect(reply.tokenInFeeAmount).toBeInstanceOf(Int);
+  // const tokenInFeeAmount = reply.tokenInFeeAmount!.toString();
+  // expect(isNumeric(tokenInFeeAmount)).toBeTruthy();
+  // expect(parseFloat(tokenInFeeAmount)).toBeGreaterThan(0);
+
+  // Split
+  expect(Array.isArray(reply.split)).toBeTruthy();
+  expect(reply.split.length).toBeGreaterThan(0);
+
+  for (const split of reply.split) {
+    expect(split.initialAmount).toBeInstanceOf(Int);
+
+    expect(Array.isArray(split.pools)).toBeTruthy();
+    expect(split.pools.length).toBeGreaterThan(0);
+
+    expect(split.tokenInDenom).toBe(atomMinimalDenom);
+
+    expect(Array.isArray(split.tokenOutDenoms)).toBeTruthy();
+    expect(split.tokenOutDenoms.length).toBeGreaterThan(0);
+  }
+
+  // Sum of all split amount should equal token in amount
+  const splitAmountSum = reply.split.reduce(
+    (acc, split) => acc.add(split.initialAmount),
+    new Int(0)
+  );
+  expect(splitAmountSum.equals(new Int(tokenInAmount))).toBeTruthy();
+
+  // name
+  expect(reply.name).toBe("tfm");
+
+  // timeMs
+  expect(isNumeric(reply.timeMs)).toBeTruthy();
+
+  // Token in fee amount fiat value
+  // expect(reply.tokenInFeeAmountFiatValue).toBeInstanceOf(PricePretty);
+
+  // const tokenInFeeAmountFiatValue = reply
+  //   .tokenInFeeAmountFiatValue!.toDec()
+  //   .toString();
+  // expect(isNumeric(tokenInFeeAmountFiatValue)).toBeTruthy();
+  // expect(parseFloat(tokenInFeeAmountFiatValue)).toBeGreaterThan(0);
+
+  // token out price
+  expect(reply.tokenOutPrice).toBeInstanceOf(PricePretty);
+
+  const tokenOutPrice = reply.tokenOutPrice!.toDec().toString();
+  expect(isNumeric(tokenOutPrice)).toBeTruthy();
+  expect(parseFloat(tokenOutPrice)).toBeGreaterThan(0);
+
+  // amount fiat value
+  expect(reply.amountFiatValue).toBeInstanceOf(PricePretty);
+
+  const amountFiatValue = reply.amountFiatValue!.toDec().toString();
+  expect(isNumeric(amountFiatValue)).toBeTruthy();
+  expect(parseFloat(amountFiatValue)).toBeGreaterThan(0);
+});
+
+async function getSortedPoolsWithVolume() {
```

Comment:
nit: consider adding a short spec outlining the details of the helper

Reply:
Added it on my last commit